src/cache.rs

```diff
@@ -341,6 +341,31 @@ impl Cache {
                 }
             })
             .collect::<Vec<String>>();
+
+        if let Some(custom_pages_dir) = custom_pages_dir {
+            let is_page = |entry: &DirEntry| -> bool {
+                let path = entry.path();
+                let extension = &path.extension().and_then(OsStr::to_str).unwrap_or("");
+                entry.file_type().is_file() && extension == &"page"
```

Comment:
This can be simplified: We dont need to convert to `str` from `OsStr`, because `OsStr` has an `impl PartialEq<&str>` already. Instead of `unwrap_or("")`, we can use `unwrap_or_default` to get the empty `OsStr`. Finally, we don't need to borrow both sides of the equality check, like so:
```suggestion
                let extension = entry.path().extension().unwrap_or_default();
                entry.file_type().is_file() && extension == "page"
```